sys/include/net/telnet.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_telnet basic Telnet server implementation
+ * @ingroup     net_ipv6
+ * @brief       Telnet server functions
+ * @{
+ *
+ * @file
+ * @brief   Telnet server ([RFC 855](https://tools.ietf.org/html/rfc855)) implementation
+ * @note    This implementation only supports a single client
+ *
+ * Telnet is entirely unencrypted! Do not use it on public networks.
+ * This is intended to aid debugging on networks that are isolated from the Internet.
+ *
+ * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+#ifndef NET_TELNET_H
+#define NET_TELNET_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   The port for the Telnet server to listen on
+ */
+#ifndef TELNET_PORT
+#define TELNET_PORT (23)
```

Comment:
```suggestion
#ifndef CONFIG_TELNET_PORT
#define CONFIG_TELNET_PORT (23)
```

And add a `KConfig` file to the modules directory, please (if at least for `CONFIG_TELNET_TCP_QUEUE_SIZE`), not sure if the port needs to be configurable.